lessons/8-intermediate-r-1/lesson-data/lesson_script.R

```diff
@@ -1,69 +1,54 @@
-
-island_dat <- island_dat |>
-  na.omit()
-
-### Calculate species-by-sex means for morphological features
-
-species_sex_means <- island_dat |>
-  group_by(species, sex) |>
-  summarize(across(ends_with("mm") | ends_with("g"), mean))
-
-#### Add number of penguins per group
-species_sex_means <- island_dat |>
-  group_by(species, sex) |>
-  summarize(across(ends_with("mm") | ends_with("g"), mean),
-            n_penguins = dplyr::n())
-
-### Convert units from metric to imperial
-
-species_sex_means_imperial <- species_sex_means |>
-  mutate(across(ends_with("mm"), ~ . * 0.03937008, .names = "{.col}_in"), 
-         across(ends_with("g"), ~ . * 0.002204623, .names = "{.col}_lb" ))
-
-#### Rename columns 
-
-species_sex_means_imperial <- species_sex_means_imperial |>
-  rename_with(~stringr::str_replace(., "mm_in", "in"), .cols = ends_with("mm_in")) |>
-  rename_with(~stringr::str_replace(., "g_lb", "lb"), .cols = ends_with("g_lb"))
-
-#### Select variables
-
-species_sex_means_imperial <- species_sex_means_imperial |>
-  select(c("n_penguins",
-           where(is.character) |
-           ends_with("in") |
-           ends_with("lb")))
-
-### Produce crosstabs
-
-library(tidyr)
-
-species_by_sex_crosstabs <- species_by_sex_means |>
-  pivot_wider(id_cols = species, names_from = sex, values_from = where(is.numeric))
-
-### Save files
-
-write.csv(species_by_sex_means, "long_means_biscoe.csv")
-write.csv(species_by_sex_crosstabs, "crosstabs_biscoe.csv")
-
-
-## then turn these into functions
-
-## start with one big function
-
-## then break into 3 (or 4): read data; clean data; get species means; change units; get crosstabs; save file.
-## then write a script chaining them together
-## then separate into functions.R and script.R
-## then run on Biscoe again to reproduce.
-## use ifelse to switch between island names.
\ No newline at end of file
+library(palmerpenguins)
+
+#### Revisit the penguins data ####
+
+penguins <- penguins
+
+head(penguins)
+
+#### Write functions to convert metric to imperial ####
+
+mm_to_inches <- function(value_in_mm) {
+  
+  value_in_inches <- value_in_mm * 0.03937008
+  
+  return(value_in_inches)
+  
+}
+
+g_to_lbs <- function(value_in_g) {
+  
+  value_in_lbs <- value_in_g * 0.002204623
+  
+  return(value_in_lbs)
+}
+
+penguins_imperial <- penguins |>
+  mutate(across(ends_with("mm"), mm_to_inches, .names = "{.col}_in"),
+         across(ends_with("g"), g_to_lbs, .names = "{.col}_lb"))
+
+#### Create a function to return penguins data in metric or imperial ####
+
+get_penguins_data <- function(units = "metric") {
+  
+  if(units == "metric") {
+    return(penguins)
+  } else if(units == "imperial") {
+    
+    penguins_imperial <- penguins |>
+      mutate(across(ends_with("mm"), mm_to_inches, .names = "{.col}_in"),
+             across(ends_with("g"), g_to_lbs, .names = "{.col}_lb"))
+    
+    return(penguins_imperial)
+  } else {
+    message("Please specify units as either 'metric' or 'imperial'!")
```

Comment:
Should this be an error? 
```suggestion
    stop("Please specify units as either 'metric' or 'imperial'!")
```